test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+void driver_test();
+void test_get_options();
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+void parse_cmd_line_and_compare(char* args[], const std::string& cmd_line, const Options& expected, const std::string& test_case);
+
+//unit test for the test driver
+void driver_test()
+{
+	test_get_options();
+}
+
+void parse_cmd_line_and_compare(const std::string& cmd_line, const Options& expected,
+	const std::string& test_case)
+{
+	//this is meant to replicate the main() parameter char* argv[]
+	//size 100 was chosen to ensure enough room for many options in a single command-line.
+	char* args[100];
+	std::vector<std::string> v = split(cmd_line, ' ');
```

Comment:
I believe a vector of `string_view` is better. I'm not sure, but with that change we might run into `const`ness of `data()`, but that can be addressed, and is worth doing so over creating a vector of strings

Reply:
I agree, `string_view` would be better to use.

Reply:
For now, the changes I've pushed do not use string_view as I'm still trying to reconcile the `const`ness of `string_view.data()` and the fact that `char* args[]` cannot be const since `get_options` requires a non-const array.

Reply:
> I believe a vector of `string_view` is better. I'm not sure, but with that change we might run into `const`ness of `data()`, but that can be addressed, and is worth doing so over creating a vector of strings

I am curious, how do we cast `const char*` to `char*`?